### ISSUE TITLE
i18n_subsites redirect to navigator lang when page lang does not match

### ISSUE THREAD

**Issue by Siltaar** [getpelican/pelican-plugins#975]
Hi,

I run a 2 langs site through Pelican. I wanted to have the site served in the user's navigator lang, so I made a small JavaScript to redirect to the navigator lang when needed.

It's here : 
https://framagit.org/Siltaar/www.meta-press.es/blob/master/theme/pelican-mp/static/js/accept-lang.js

It requires a link to the wanted navigator lang available on the page and only tries to do something when visitor is landing on the website (no referrer or different from considered website). So if the visitor choose another lang, it will stay on.

Works from default_lang to different navigator lang, and from different page lang to default_lang.

Tested with Firefox 57 and Chrome 63.

**Comment by justinmayer** [comment]
@Siltaar: Is this something you would consider adding as an optional feature for the i18n_subsites plugin?

**Comment by Siltaar** [comment]
Yes :-)

I think that redirecting the user toward its navigator lang is desirable.
I suggested a way to achieve this. Maybe it's possible to redirect earlier ?

My script is relying on the availability of lang-chooser links in the page, else it will do the job.

**Comment by justinmayer** [comment]
Wonderful. Looking forward to your pull request. 😁

**Comment by mwallnoefer** [comment]
Yes, also I am interested to see this implemented. Please update me when it's here.

**Comment by Siltaar** [comment]
I pushed what look like needed in my fork and it's stacked over my current pull request for pelican-plugins.
https://github.com/getpelican/pelican-plugins/pull/1004

The two things are good and maybe it could be accepted all together…

**Comment by Siltaar** [comment]
@mwallnoefer a new PR has been created with only this feature in it.
